src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -306,6 +332,34 @@ export const getTopNavConfig = (
           },
         ]
       : []),
+    ...(originatingApp === 'dashboards'
```

Comment:
It seems to me that we'd like to show the cancel button when editing from canvas too, since the navigateTo methods should be able to work in either case!